dace/frontend/python/replacements.py

```diff
@@ -127,7 +127,15 @@ def _reduce(sdfg: SDFG,
                                            {})
         input_memlet = Memlet.simple(inarr, input_subset)
         output_shape = None
-        if axis is None:
+
+        # check if we are reducing along all axes
+        if axis is not None and len(axis) == len(input_subset.size()):
```

Comment:
Do you mean `is None or...`?

Reply:
No I mean `is not None`. The `is None` case is handled on L138